src/basic/efivars.c

```diff
@@ -30,48 +30,22 @@
 #define EFI_N_RETRIES_TOTAL 25
 #define EFI_RETRY_DELAY (50 * USEC_PER_MSEC)
 
-char* efi_variable_path(sd_id128_t vendor, const char *name) {
-        char *p;
-
-        if (asprintf(&p,
-                     "/sys/firmware/efi/efivars/%s-" SD_ID128_UUID_FORMAT_STR,
-                     name, SD_ID128_FORMAT_VAL(vendor)) < 0)
-                return NULL;
-
-        return p;
-}
-
-static char* efi_variable_cache_path(sd_id128_t vendor, const char *name) {
-        char *p;
-
-        if (asprintf(&p,
-                     "/run/systemd/efivars/%s-" SD_ID128_UUID_FORMAT_STR,
-                     name, SD_ID128_FORMAT_VAL(vendor)) < 0)
-                return NULL;
-
-        return p;
-}
-
 int efi_get_variable(
-                sd_id128_t vendor,
-                const char *name,
+                const char *variable,
```

Comment:
I am not sure I like this too much I must say. Sounds like the wrong place to drop abstraction and to optimize. This is not inner loop/performance relevant stuff. And the `<lowercaseuuid>-<varname>` naming scheme is an efivarfs thing, i.e. of the Linux API file system implementation for EFI. it's not something the EFI spec would declare, it processes the UUID only in binary form...

Taking the binary UUID has the benefit that the ambiguity around uppercase/lowercase UUIDs is not exposed. Once you change the APIs to accept things in string form you export that whole mess everywhere though.

Moreover, any code that actually intends to pass a variable uuid/varname pair now has to jump through extra hoops to convert that int the right string, as the macro you defined can't really be used for that...

Hence, not convinced this is really any improvment.  But then again, I am not strongly against it, so if you really want this, fine. But please add a comment somewhere that makes clear the `<lowercaseuuid>-<varname>` thing is

Reply:
I'll add the comments.

If this was a public interface, I fully agree that we shouldn't encode the details in the api like this. But as long as it's purely internal, we can cut some corners. If it ever turns out that we need a different format, we can adjust/undo those changes.